dyn/tm/session.py

```diff
@@ -175,4 +181,89 @@ def __str__(self):
         """str override"""
         header = super(DynectSession, self).__str__()
         return header + force_unicode(': {}, {}').format(self.customer,
-                                                         self.username)
\ No newline at end of file
+                                                         self.username)
+
+
+class DynectMultiSession(DynectSession):
+
+    def __init__(self, customer, username, password, host='api.dynect.net',
+                 port=443, ssl=True, api_version='current', auto_auth=True,
+                 key=None, history=False, proxy_host=None, proxy_port=None,
+                 proxy_user=None, proxy_pass=None):
+
+        self._open_sessions = {}
+
+        super(DynectMultiSession, self).__init__(customer, username, password, host=host,
+                                                 port=port, ssl=ssl, api_version=api_version, auto_auth=auto_auth,
+                                                 key=key, history=history, proxy_host=proxy_host, proxy_port=proxy_port,
+                                                 proxy_user=proxy_user, proxy_pass=proxy_pass)
+
+    def _handle_error(self, uri, method, raw_args):
+        """Handle the processing of a connection error with the api"""
+        # Our token is no longer valid because our session was killed
+        self._token = None
+        # Need to force a re-connect on next execute
+        self._conn.close()
+        self._conn.connect()
+        # Need to get a new Session token and update the open session
+        self.authenticate()
+        # Then try the current call again and Specify final as true so
+        # if we fail again we can raise the actual error
+        return self.execute(uri, method, raw_args, final=True)
+
+    def __add_open_session(self):
+        """Add new open session to hash of open sessions"""
+        self._open_sessions[self.username] = {
+            'user_name': self.username,
+            'password': self.password,
+            'customer_name': self.customer,
+            'token': self._token
+        }
+
+    def get_open_sessions(self):
+        return self._open_sessions
+
+    def set_active_session(self, username):
+        """Set the active session from the hash of open sessions"""
+        if username in self._open_sessions:
+            self.username = self._open_sessions[username]['user_name']
+            self.password = self._open_sessions[username]['password']
+            self.customer = self._open_sessions[username]['customer_name']
+            self._token = self._open_sessions[username]['token']
+        else:
+            raise ValueError("No open sessions for {0}".format(username))
+
+    def new_user_session(self, customer, username, password):
```

Comment:
We shouldn't really require customer be included should we? I can see how switching session between customers could become troublesome... Are we expecting end users to use this to flip flop between different customer accounts?